src/opower/utilities/nationalgrid.py

```diff
@@ -0,0 +1,72 @@
+"""National Grid Upstate NY utility implementation.
+
+This module handles the authentication and API interactions for National Grid.
+It uses OAuth 2.0 with PKCE for secure authentication.
+"""
+
+import logging
+from typing import Optional
+
+import aiohttp
+
+from .base import UtilityBase
+from .oidchelper import async_auth_oidc
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class NationalGrid(UtilityBase):
```

Comment:
Rename file and class since this is specific to NY Upstate.

Reply:
b5676c1